### PR TITLE
Add rattler-build recipe for Zig-built conda-launchers

### PR DESCRIPTION
**Pull request by chawyehsu**
To address #4 ~, requires #8 to trigger the CI~

### CODE REVIEW

File: conda-launchers/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
+context:
+  name: conda-launchers
+  version: "25.0.0"
```
Comment (chawyehsu) [inline conda-launchers/recipe.yaml:3]: Not sure what version number is properly used here.

File: conda-launchers/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          - if not exist %PREFIX%\Scripts\cli-arm64-script.py exit 1
+          # gui
+          - if not exist %PREFIX%\Scripts\gui-32.exe exit 1
+          - if not exist %PREFIX%\Scripts\gui-32-script.pyw exit 1
+          - call "%PREFIX%\Scripts\gui-32.exe"
+          - if not exist %PREFIX%\Scripts\gui-64.exe exit 1
+          - if not exist %PREFIX%\Scripts\gui-64-script.pyw exit 1
+          - call "%PREFIX%\Scripts\gui-64.exe"
+          - if not exist %PREFIX%\Scripts\gui-arm64.exe exit 1
+          - if not exist %PREFIX%\Scripts\gui-arm64-script.pyw exit 1
```
Comment (chawyehsu) [inline conda-launchers/recipe.yaml:64]: can't run arm64 executable on x64 vm, so no tests for arm64 launchers here.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -4,9 +4,7 @@ on:
   push:
     branches:
       - main
-  pull_request:
-    branches:
-      - main
+  pull_request_target:
```
Comment (jaimergp) [inline .github/workflows/test.yml:7]: We don't need this. Let's revert.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -85,3 +83,50 @@ jobs:
...
           echo "  conda install -c ${ANACONDA_ORG_CHANNEL}/label/${ANACONDA_ORG_LABEL} conda-launchers"
+
+  rattler_build_recipe:
+    name: Rattler build recipe
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (jaimergp) [inline .github/workflows/test.yml:94]:
```suggestion
```

File: conda-launchers/recipe.yaml
Comment (jaimergp) [inline conda-launchers/recipe.yaml:1]: Can't these files coexist under `recipe` together with meta.yaml?

File: conda-launchers/recipe.yaml (hunk 1 above)
Comment (jaimergp) [reply conda-launchers/recipe.yaml:3]: For now let's use the same as in `meta.yaml`

File: src/launcher.manifest
Comment (jaimergp) [inline src/launcher.manifest:1]: Why is this needed? Does it break the other builds?

Comment (chawyehsu) [reply src/launcher.manifest:1]: Zig will not be happy and will not build if the manifest file extension is not `.manifest`, the [official ext name](https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#file-name).

I've updated the `build.sh` to reflect this name change. If you don't like this there's another way by duplicating the manifest file leaving the .xml one unchanged, which creates redundant code.

File: conda-launchers/recipe.yaml
Comment (chawyehsu) [reply conda-launchers/recipe.yaml:1]: It's possible, I just don't like mixing two different types of recipe code here.

File: src/launcher.manifest
Comment (jaimergp) [reply src/launcher.manifest:1]: Ok, that's fine then. Thanks for the clarification!

**Comment by chawyehsu** [comment]
Great, all CI checks have passed, and we got our first [CI build](https://github.com/conda/conda-launchers/actions/runs/11742117875).

File: conda-launchers/recipe.yaml (hunk 2 above)
Comment (jaimergp) [reply conda-launchers/recipe.yaml:64]: You could use platform selectors here if you want.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -85,3 +85,49 @@ jobs:
...
+        with:
+          name: conda-launchers-builds
+          path: |
+            ./output/noarch/conda-launchers*.conda
+
+      - name: Upload to Anaconda
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        env:
+          ANACONDA_OWNER: conda-canary
+          ANACONDA_CHANNEL: dev
```
Comment (jaimergp) [inline .github/workflows/test.yml:122]: Can we use a different label here so the two versions don't clash? 
```suggestion
          ANACONDA_CHANNEL: dev-zig
```

File: conda-launchers/recipe.yaml
Comment (jaimergp) [reply conda-launchers/recipe.yaml:1]: If you want to keep both, can we rename the directory to `recipe-zig` then?

File: conda-launchers/build.bat [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+if %PKG_NAME% == conda-launchers_win-arm64 (
+    set ZIG_TARGET=aarch64-windows-gnu
+    set EXE_TARGET=arm64
+)
+
+@rem build cli launcher
+zig build -Doptimize=ReleaseSmall -Dtarget=%ZIG_TARGET% -Dgui=false --prefix-exe-dir "%PREFIX%\Scripts"
+@rem build gui launcher
```
Comment (jaimergp) [inline conda-launchers/build.bat:26]:
```suggestion
if %ERRORLEVEL% neq 0 exit 1

@rem build gui launcher
```

File: conda-launchers/build.bat [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+zig build -Doptimize=ReleaseSmall -Dtarget=%ZIG_TARGET% -Dgui=true --prefix-exe-dir "%PREFIX%\Scripts"
+
+if %ERRORLEVEL% neq 0 exit 1
+
+@rem install launcher scripts
+cd "%PREFIX%\Scripts"
+(
+echo print^("cli-%EXE_TARGET%.exe successfully launched the accompanying Python script"^)
+)> "cli-%EXE_TARGET%-script.py"
+
```
Comment (jaimergp) [inline conda-launchers/build.bat:36]:
```suggestion
if %ERRORLEVEL% neq 0 exit 1

```

**Comment by chawyehsu** [comment]
I've updated the PR as per the review, please trigger the CI and let me know if there's any tweak needed.

**Comment by jaimergp** [comment]
Thanks!
